mycli/output_formatter.py

```diff
@@ -0,0 +1,232 @@
+    """Align decimals to decimal point
+    >>> for i in align_decimals([[Decimal(1)], [Decimal('11.1')], [Decimal('1.1')]], [])[0]: print(i[0])
+     1
+    11.1
+     1.1
+    """
+    pointpos = len(data[0]) * [0]
+    for row in data:
+        for i, v in enumerate(row):
+            if isinstance(v, Decimal):
+                v = encodingutils.text_type(v)
+                pointpos[i] = max(intlen(v), pointpos[i])
+    results = []
+    for row in data:
+        result = []
+        for i, v in enumerate(row):
+            if isinstance(v, Decimal):
+                v = encodingutils.text_type(v)
+                result.append((pointpos[i] - intlen(v)) * " " + v)
+            else:
+                result.append(v)
+        results.append(result)
+    return results, headers
+
+
+def quote_whitespaces(data, headers, quotestyle="'", **_):
+    """Quote whitespace
+    >>> for i in quote_whitespaces([["  before"], ["after  "], ["  both  "], ["none"]], [])[0]: print(i[0])
+    '  before'
+    'after  '
+    '  both  '
+    'none'
+    >>> for i in quote_whitespaces([["abc"], ["def"], ["ghi"], ["jkl"]], [])[0]: print(i[0])
+    abc
+    def
+    ghi
+    jkl
+    """
+    quote = len(data[0]) * [False]
+    for row in data:
+        for i, v in enumerate(row):
+            v = encodingutils.text_type(v)
+            if v.startswith(' ') or v.endswith(' '):
+                quote[i] = True
+
+    results = []
+    for row in data:
+        result = []
+        for i, v in enumerate(row):
+            quotation = quotestyle if quote[i] else ''
+            result.append('{quotestyle}{value}{quotestyle}'.format(
+                quotestyle=quotation, value=v))
+        results.append(result)
+    return results, headers
+
+
+def tabulate_wrapper(data, headers, table_format=None, missing_value='', **_):
+    """Wrap tabulate inside a standard function for OutputFormatter."""
+    return tabulate(data, headers, tablefmt=table_format,
+                    missingval=missing_value, disable_numparse=True)
+
+
+def csv_wrapper(data, headers, delimiter=',', **_):
+    """Wrap CSV formatting inside a standard function for OutputFormatter."""
+    with contextlib.closing(StringIO()) as content:
+        writer = csv.writer(content, delimiter=str(delimiter))
+
+        writer.writerow(headers)
+        for row in data:
+            writer.writerow(row)
+
+        return content.getvalue()
+
+
+def terminal_tables_wrapper(data, headers, table_format=None, **_):
+    """Wrap terminaltables inside a standard function for OutputFormatter."""
+    if table_format == 'ascii':
+        table = terminaltables.AsciiTable
+    elif table_format == 'single':
+        table = terminaltables.SingleTable
+    elif table_format == 'double':
+        table = terminaltables.DoubleTable
+    elif table_format == 'github':
+        table = terminaltables.GithubFlavoredMarkdownTable
+    else:
+        raise ValueError('unrecognized table format: {}'.format(table_format))
```

Comment:
Whenever we see a series of `if/elif/else` section it is better to replace it with a dictionary as follows: 

```
table_formats = { 
    'ascii': terminaltables.AsciiTable, 
    'single': terminaltables.SingleTable,  
    .....
    }

try:
    table = table_formats[table_format]
except KeyError:
    raise ValueError('unrecognized table format: {}'.format(table_format))
```